ttnn/cpp/ttnn/tensor/types.cpp

```diff
@@ -160,6 +160,17 @@ const LegacyShape LegacyShape::without_padding() const {
     return LegacyShape(shape_without_padding);
 }
 
+ttnn::SimpleShape LegacyShape::logical_shape() const
+{
+    const LegacyShape logical = without_padding();
+
+    std::vector<uint32_t> values(rank());
+    for (size_t i = 0; i < values.size(); i++) {
```

Comment:
I wanted to suggest std::copy here but some quickbench shows that perf is noticeably worse... I was shocked that copy_n didn't match or exceed the perf of a raw loop... Disappointed in the library and compiler :'(
Although intent is clearer with std::copy

<img width="972" alt="image" src="https://github.com/user-attachments/assets/cc798e9d-0abd-4576-8dc6-f98deb7a3f31">

Reply:
this flow should give comparable perf
```
vector
reserve
copy
```